compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -678,6 +678,117 @@ def IREELinalgExt_AttentionOp : IREELinalgExt_Op<"attention",
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// Im2col
+//===----------------------------------------------------------------------===//
+
+def IREELinalgExt_Im2colOp : IREELinalgExt_Op<"im2col",
+    [DeclareOpInterfaceMethods<ReifyRankedShapedTypeOpInterface>]> {
+  let summary = "Im2col operation for convolutions";
+  let description = [{
+    Im2col op for convolutions. The operation performs a transformation on the
+    input to convert it from a convolution input to an equivalent gemm input.
+    The op is defined by its input, output, some conv metadata, and some
+    indexing metadata. The `strides`, `dilations`, and `kernel_size` are taken
+    from the convolution from which this op is generated, and they define how
+    the input operand is indexed when the operation is decomposed. The shape of
+    the output should be `tensor<BxMxK>`, and the `m_pos`, `k_pos`, and
```

Comment:
Nit: I understand what you are trying to do with `m_pos`, `k_pos` and `batch_pos`, but it is more intuitive to work from the conv spec (i.e. `N`, `C`, `H`, `W`) to gemm spec (`B`, `M`, `K`), . IIUC , 'N' always maps to `B` of the gemm, `C` always maps to `K` and [`H`, `W`] always maps to `M`. so you can just record the n, h, w, and c of input tensor.

Reply:
The reason I chose to use the `BMK` naming instead of `NHWC` is that the `NCHW` dimensions do not always map the same way to `BMK`. For example, in a depthwise conv, the `C` dimension is a batch dimension. I think it's better to have a mapping from input dimensions to output dimensions, rather than having the mapping be implicit with the type of input dimension. This way it is not tied to specific conv layouts, and it is more generally mapping some convolved access pattern to a matmul access pattern.